src/gui/gui.c

```diff
@@ -3275,11 +3275,11 @@ static void GUI_StrategicMap_PrepareRegions(uint16 campaignID)
 static void GUI_StrategicMap_ShowProgression(uint16 campaignID)
 {
 	char key[10];
-	char category[10];
+	char category[11];
 	char buf[100];
 	uint16 i;
 
-	sprintf(category, "GROUP%d", campaignID);
+	sprintf(category, "GROUP%hu", campaignID);
```

Comment:
better change to snprintf()

Reply:
instead of increasing category buffer size I mean.

Reply:
In this case, gcc sees that a %d (int) on x86_64 is 32 bits (signed) and %d can produce at most 11 bytes == strlen("-2147483648"). A half unsigned int (%hu), which will output at most 5 bytes == strlen("65535") fits the output buffer.

Since we have a reoccurring problem with s*printf and length of output buffer and all(?) cases are similar, we try to build a 8.3 file name based on a 5 bytes base, 3 bytes number and finally the extension, maybe it would be worthwhile to build our own function to build these and avoid the complication of s*printf altogether.

Reply:
I vote for snprintf(), easiest fix and will work on any sizes of int in the future as well